### PR TITLE
* slice filter: Return empty array if `start >= end`

### LINKED ISSUE #702: if `start > end` then `slice` filter panicks
Hello!

```
{{ [1, "foo", 1, "true", "FOO", true, 2] | slice(start=4, end=2) }}
```

panick:
```
thread 'thread '<unnamed>thread '<unnamed>' panicked at 'slice index starts at 4 but ends at 2' panicked at 'slice index starts at 4 but ends at 2<unnamed>' panicked at '', thread 'slice index starts at 4 but ends at 2<unnamed>' panicked at 'slice index starts at 4 but ends at 2', /home/rootkea/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.15.0/src/builtins/filters/array.rs/home/rootkea/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.15.0/src/builtins/filters/array.rs:285:8
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
:285:8
', ', /home/rootkea/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.15.0/src/builtins/filters/array.rs:285:8
thread 'thread '<unnamed>' panicked at 'slice index starts at 4 but ends at 2', /home/rootkea/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.15.0/src/builtins/filters/array.rs:285:8
/home/rootkea/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.15.0/src/builtins/filters/array.rs:285:8
thread '<unnamed>' panicked at 'slice index starts at 4 but ends at 2thread '<unnamed>' panicked at 'slice index starts at 4 but ends at 2', /home/rootkea/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.15.0/src/builtins/filters/array.rs:285:8
thread '<unnamed>' panicked at 'slice index starts at 4 but ends at 2', /home/rootkea/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.15.0/src/builtins/filters/array.rs:285:8
thread '<unnamed>' panicked at 'slice index starts at 4 but ends at 2', /home/rootkea/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.15.0/src/builtins/filters/array.rs:285:8
<unnamed>', /home/rootkea/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.15.0/src/builtins/filters/array.rs:285:8
' panicked at 'thread 'slice index starts at 4 but ends at 2', /home/rootkea/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.15.0/src/builtins/filters/array.rs:285:8
<unnamed>' panicked at 'slice index starts at 4 but ends at 2', /home/rootkea/.cargo/registry/src/github.com-1ecc6299db9ec823/tera-1.15.0/src/builtins/filters/array.rs:285:8
```

Interestingly, if `start == end` empty array is returned. So what to do when `start > end`?
1. Return empty array OR
2. throw error

### PR DESCRIPTION
**Pull request by rootkea**
Fixes #702

### CODE REVIEW

**Comment by rootkea** [comment]
How do I use this compiled tera to build updated zola for testing purpose?

**Comment by Keats** [comment]
> How do I use this compiled tera to build updated zola for testing purpose?

Replace all the `tera = "1"` in all Cargo.toml of Zola with `tera = { path = "../my/path/to/local/tera" }`

File: src/builtins/filters/array.rs [hunk 1]
```diff
@@ -282,6 +277,11 @@ pub fn slice(value: &Value, args: &HashMap<String, Value>) -> Result<Value> {
         end = arr.len();
     }
 
+    // Not an error, but returns an empty Vec
+    if start >= end {
+        return Ok(Vec::<Value>::new().into());
+    }
+
```
Comment (Keats) [inline src/builtins/filters/array.rs:284]: Can you add a test for it?

Comment (rootkea) [reply src/builtins/filters/array.rs:284]: Done!

BTW, I tried adding `(make_args(Some(-1), Some(-4.0)), vec![])` to tests but it gave compile error "error[E0600]: cannot apply unary operator `-` to type `usize`".

**Comment by rootkea** [comment]
Thanks!

**Pull request merged**